core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -320,14 +321,15 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         .filter { entry =>
           try {
             val prevFileSize = fileToAppInfo.get(entry.getPath()).map{_.fileSize}.getOrElse(0L)
+            fs.access(entry.getPath, FsAction.READ)
```

Comment:
So, this API actually calls `fs.getFileStatus()` which is a remote call and completely unnecessary in this context, since you already have the `FileStatus`.

You could instead directly do the check against the `FsPermission` object (same way the `fs.access()` does after it performs the remote call).